realscale.py

```diff
@@ -59,16 +82,83 @@ def calc_scale_center(self, p1x, p1y, p2x, p2y):
         if line_x < pagecenter_x:
             # Left hand side
             if line_y < pagecenter_y:
-                scale_center = [minx,miny] # UL
+                scale_center = (minx,miny) # UL
             else:
-                scale_center = [minx,maxy] # LL
+                scale_center = (minx,maxy) # LL
         else: # Right hand side
             if line_y < pagecenter_y:
-                scale_center = [maxx,miny] # UR
+                scale_center = (maxx,miny) # UR
             else:
-                scale_center = [maxx,maxy] # LR
+                scale_center = (maxx,maxy) # LR
         #inkex.debug("%s  %s,%s" % (scale_center, pagecenter_x*2, pagecenter_y*2))
         return scale_center
+    
+    def create_ruler(self, parent, width, pos, value, drawnscale):
+        """ Draw Scale rule
+            - Position above user's straightline.
+            - Ruler shows two units together. First one cut into 5
+            pos is a tuple e.g. (0,0)
+            Todo:
+            - Add magnification e.g. 2:1 for small drawings
+            - Override scale if result is > page size
+        """
+        " Ruler is always 2 units long with 5 divs on LHS "
+        # Draw two boxes next to each other. Top half of RHS is filled black
+        line_width = self.getUnittouu('0.25 mm')
+        box_height = max(width/15, self.getUnittouu('2 mm'))
+        font_height = 8
+        White = '#ffffff'
+        Black = '#000000'
+        t = 'translate' + str(pos)
+        group = inkex.etree.SubElement(parent, 'g', {inkex.addNS('label','inkscape'):"scale_rule", 'transform':t})
+        # RHS box
+        boxR = {'x':'0.0', 'y':'0.0',
+                'width':str(width), 'height':str(box_height),
+                'style':'fill:%s;stroke:%s;stroke-width:%s;stroke-opacity:1;fill-opacity:1' % (White, Black, line_width)}
+        inkex.etree.SubElement(group, 'rect', boxR)
+        # top half black
+        boxRf = {'x':'0.0', 'y':'0.0',
+                 'width':str(width), 'height':str(box_height/2),
+                 'style':'fill:%s;stroke:%s;stroke-width:%s;stroke-opacity:1;fill-opacity:1' % (Black, Black, line_width)}
+        inkex.etree.SubElement(group, 'rect', boxRf)
+        # LHS
+        boxL = {'x':str(-width), 'y':'0.0',
+                'width':str(width), 'height':str(box_height),
+                'style':'fill:%s;stroke:%s;stroke-width:%s;stroke-opacity:1;fill-opacity:1' % (White, Black, line_width)}
+        inkex.etree.SubElement(group, 'rect', boxL)
+        # staggered black fills on LHS
+        start = -width
+        for i in range(5):
+            boxRf = {'x':str(start), 'y':str((i+1)%2 * box_height/2), 
+                     'width':str(width/5), 'height':str(box_height/2),
+                     'style':'fill:%s;stroke:%s;stroke-width:%s;stroke-opacity:1;fill-opacity:1' % (Black, Black, line_width)}
```

Comment:
I don't think they should have a stroke. The strokes overlap for me and make it hard to distinguish where the split is between a box and the next one below. Somehow they do not count towards the width. 1mm becomes 1.25 mm.

![bildschirmfoto_2016-04-27_18-56-46](https://cloud.githubusercontent.com/assets/3240233/14860588/d9154a9c-0ca9-11e6-92d7-825d373f8c80.png)